sql/src/main/java/io/druid/sql/calcite/planner/DruidRexExecutor.java

```diff
@@ -84,7 +84,7 @@ public void reduce(
         if (sqlTypeName == SqlTypeName.BOOLEAN) {
           literal = rexBuilder.makeLiteral(exprResult.asBoolean(), constExp.getType(), true);
         } else if (sqlTypeName == SqlTypeName.DATE) {
-          if (!constExp.getType().isNullable() && exprResult.isNull()) {
+          if (!constExp.getType().isNullable() && exprResult.isNumericNull()) {
```

Comment:
Why this?

Reply:
It is possible for an expression to have a non-null String value but it can return null when parsed as a primitive long/float/double.
ExprEval.isNumericNull checks whether the parsed primitive value is null or not.
Added comment to code.